consumption.tex

```diff
@@ -0,0 +1,74 @@
+
+\begin{itemize}
+  \item $\textsf{submit-event-proof}(\pi, e)$: A prover calls this function while putting down a collateral $z$ with an event $e$ and a proof of it $\pi$. The function determines if the proof provided is valid by making use of the \textsf{verify} method. If the proof is valid, a contestation period of $k$ blocks begins where other provers may submit a contesting proof.
+  \item $\textsf{submit-contesting-proof}(\pi^*, e)$: A prover calls this function if they have a contesting proof for the proof provided previously for event $e$. If through the use of \textsf{verify} it is discovered that the original prover was dishonest, her collateral gets transferred to the caller of this function and the event is discarded.
+  \item $\textsf{finalize-event}(e)$: A prover calls this function when the contestation period has passed for their event $e$. If there has been no successful contesting prover in the meantime, their event is saved in the set of finalized events on the smart contract.
+  \item $\textsf{event-exists}(e)$: A user of the contract calls this function to determine whether an event $e$ has been proven and finalized.
+\end{itemize}
+
+\import{./}{algorithms/alg.verify-tx.tex}
+
+Before looking at how the verification works for the whole proof $\beta$ we focus on the part of the proof pertaining to the transaction, specifically $\alpha = \sf(tx, \tau, b)$. Verifying this part of the proof for some event $e$ is straightforward by making use of the blockchain transaction protocol in Definition~\ref{def:blockchain-tx}. We ensure that the transaction \textsf{tx} indeed contains the value transfer claimed in the provided event $e$, and that the transaction is really contained in the block provided. The function, \textsf{verify-tx}, can be seen in detail in Algorithm~\ref{alg.verify-tx}.
+
+Now that the issue of verifying the transaction is covered, we move our attention to the issue of verifying that the provided block \textsf{b} really belongs to the best source blockchain. For this we make use of NIPoPoWs. The block connection proof is a NIPoPoW for the block which includes the burn transaction.
+
+\import{./}{algorithms/alg.verify-event-nipopow.tex}
+
+The NIPoPoW event verification function, illustrated in Algorithm~\ref{alg.verify-event-nipopow}, first verifies the transaction part of the proof by making use of \textsf{verify-tx} and subsequently delegates to the NIPoPoW verification function \textsf{verify-infix}. In case of a contesting proof, the parameter $\mathcal{P}$ passed to the verification function contains both the original and the contesting proofs.
```

Comment:
I suggest we drop all references to infix and use NIPoPoWs as a black box which can answer the question: Is block $b$ confirmed in the source blockchain?

Reply:
so you don't want to talk at all about contestation?